### ISSUE TITLE
Validation messages for the login modal

### ISSUE THREAD

**Issue by SidharthBansal** [publiclab/plots2#4250]
- [ ] Login form validations are working fine for incorrect input fields(testing)
Unable to test, redirected to PL stable

**Comment by SidharthBansal** [comment]
Refer #4210

**Comment by SidharthBansal** [comment]
Oorjit's comment  
>Login modal validations are working fine for incorrect input fields(testing)
You're redirected to PL login page
  Login form validations are working fine for incorrect input fields(testing)
You're redirected to unstable PL login page with an error "There is nobody in our system by that name, are you sure you have the right username?"

**Comment by oorjitchowdhary** [comment]
I can help out here should you need any help... @sidharthbansal

**Comment by SidharthBansal** [comment]
We need the error messages to be displayed on the modal. No redirect to the /login page.

**Comment by oorjitchowdhary** [comment]
Ohk.. I'll try to look into this then.. seems important

**Comment by oorjitchowdhary** [comment]
@SidharthBansal this one will require changes at `app/views/user_sessions/_new.html.erb`?

**Comment by SidharthBansal** [comment]
I hope yes.

**Comment by SidharthBansal** [comment]
@JonathanXu1 and other GCI students if any task is left for reward then please submit it on the GCI platform. If I can approve it, I will approve it immediately.

**Comment by SidharthBansal** [comment]
@okonek @kevinzluo @dinaelhanan @oorjitchowdhary and others

**Comment by oorjitchowdhary** [comment]
Hi.. 2 are left but one is incorrect.. Can I submit that one

**Comment by SidharthBansal** [comment]
I mean the correct ones.

**Comment by SidharthBansal** [comment]
submit the correct ones now.

**Comment by SidharthBansal** [comment]
Please do it 5 minutes remaining. I am not receiving any emails from you guys.

**Comment by oorjitchowdhary** [comment]
Yes.. Is this the line which is to be changed?
https://github.com/publiclab/plots2/blob/eeed6645474a356cdd62fdd9e7c8467935faef1f/app/views/user_sessions/_new.html.erb#L5

**Comment by SidharthBansal** [comment]
This is the line to display the error messages on the login page. You need to bring in some clever insight so that the page will not redirect to /login, instead this line is rendered on the modal itself.

**Comment by SidharthBansal** [comment]
Does this makes sense?

**Comment by oorjitchowdhary** [comment]
Yes.. I get it.. I'm looking into it..

**Comment by oorjitchowdhary** [comment]
@SidharthBansal

**Comment by SidharthBansal** [comment]
Wait on this one. We have some other PR depending on these validations.
Stuff is interlinked in this OAuth Milestone. Sorry for the trouble.

**Comment by SidharthBansal** [comment]
@kevinzluo may be you will be willing to work on this?
As it is related to your pr #3984 
@oorjitchowdhary is unavailable for this month as he is having his exams. So you may take this up.

**Comment by kevinzluo** [comment]
I have my exams next week but I would love to work on it as soon as possible!

**Comment by SidharthBansal** [comment]
Is anyone working on this? @kevinzluo @oorjitchowdhary ?
I think @kevinzluo might be working on this as it is continuation of his PR.
Thanks

**Comment by SidharthBansal** [comment]
@kunalvaswani123 you can work on this one as @kevinzluo and @oorjitchowdhary are busy with some other PRs.
@kunalvaswani123 if you want I can assign more issues to you. Thanks for help.

**Comment by kunalvaswani123** [comment]
sure I'll do it after I complete tests for https://github.com/publiclab/plots2/pull/4386 ...
Yeah I am ok with more issues .. Thanks

**Comment by SidharthBansal** [comment]
You can claim 3 issues at a time. So, feel free to take this one also. It is related to the work you are doing at Public Labs. You can check out the current Kevin's PR for help. He has done a lot of work for the sign up.

**Comment by kunalvaswani123** [comment]
Sorry @SidharthBansal I didn't know about it .. I'll start working on this issue..

**Comment by kunalvaswani123** [comment]
@SidharthBansal just a clarification I need to redirect all the error messages back to login modal .. instead of login page it is directing to ?

**Comment by SidharthBansal** [comment]
If user is filling wrong details at the login modal then modal should show
errors. If page then page should show errors.

**Comment by SidharthBansal** [comment]
Kunal you can check the Kevin's PRS.
That may help you further.
One way is ajax as suggested by Kevin. Other way (easier) to show rails
validation error at top of signup and login modal as we have on respective
pages.

**Comment by kunalvaswani123** [comment]
@SidharthBansal I'll look into it.. Thanks..
Thanks @kevinzluo for your idea..

**Comment by SidharthBansal** [comment]
Now we are adding JS validations instead of Rails as Rails configuration says to redirect to the main page.

**Comment by kunalvaswani123** [comment]
I am sorry, I didn't reference a PR till now, I had 3 deadlines last week. I'll reference a PR as soon as I can..

**Comment by kunalvaswani123** [comment]
@SidharthBansal I have left a comment in https://github.com/publiclab/plots2/pull/4568 regarding this issue.. if you could please see to it .. Thanks

**Comment by SidharthBansal** [comment]
@VladimirMikulic I think you can take this issue also for JS validations. You can check the correct format of username or email AND password.
This can be a medium task for GCI. Would you like to solve this?

**Comment by VladimirMikulic** [comment]
@SidharthBansal it would be my pleasure.

**Comment by SidharthBansal** [comment]
Oh great. Is there any issue uploaded for similar task?

**Comment by VladimirMikulic** [comment]
Currently no.

**Comment by SidharthBansal** [comment]
Once it is complete and merged then ping me. I will upload the task and
share the link with you

**Comment by VladimirMikulic** [comment]
No problem :)

**Comment by SidharthBansal** [comment]
Published on GCI dashboard

**Comment by SidharthBansal** [comment]
@VladimirMikulic did you move from one pr to another pr?
I am confused. Sorry

**Comment by VladimirMikulic** [comment]
@SidharthBansal yes, I did. The one that I closed was messy, so I've decided to open a new one.
I apologise for the confusion.
